### DISCUSSION TITLE
Compile time slower in version 0.11.7-86-18d144

### DISCUSSION THREAD

**Discussion by romain-gilles-ultra** [com-lihaoyi/mill#3190] [Development]
Hi Mill,
I updated my CI with the mill version `0.11.7-86-18d144` and noticed a performance decrease in the compilation time in parallel mode.
I was on version `0.11.7-70-654f58` before that. Here are the numbers:

### Before update
Version: `0.11.7-70-654f58`
```
./mill __.compile => 5m 36s
./mill -j 4 __.compile => 2m 40s
```

### After update
Version: 0.11.7-86-18d144
```
./mill __.compile => 5m 34s
./mill -j 4 __.compile => 3m 28s
```

### Issue 
The compile time between `0.11.7-70-654f58`  and `0.11.7-86-18d144` increased by 30% on my build in parallel mode

**Comment by lefou** [comment]
https://github.com/com-lihaoyi/mill/compare/654f58...18d144

**Comment by lefou** [comment]
```
> echo -n "0.11.7-70-654f58" > .config/mill-version && mill -i -j 0 __.prepareOffline + clean __.compile && time mill -i -j 4 __.compile
...
real    1m18,300s
user    12m54,230s
sys     0m25,926s

> echo -n "0.11.7-86-18d144" > .config/mill-version && mill -i -j 0 __.prepareOffline + clean __.compile && time mill -i -j 4 __.compile
...
real    1m26,116s
user    13m39,047s
sys     0m24,853s

> echo -n "0.11.7-70-654f58" > .config/mill-version && mill -i -j 0 __.prepareOffline + clean __.compile && time mill -i -j 4 __.compile
...
real    1m13,489s
user    12m7,180s
sys     0m19,427s

> echo -n "0.11.7-86-18d144" > .config/mill-version && mill -i -j 0 __.prepareOffline + clean __.compile && time mill -i -j 4 __.compile
...
real    1m13,100s
user    12m11,088s
sys     0m19,742s

```

I can't notice any significant difference on my laptop when compiling the full Mill repo. I think the most relevant change is:

* https://github.com/com-lihaoyi/mill/pull/2899

Beside some improvements and changes regarding over- and undercompilation, zinc 1.10 introduced a new analysis file store which is itself using multiple threads. I already limited its concurrency to 8 threads (which was chosen arbitrarily). But it could be relevant, given the different OS and physical core count.
